activerecord/lib/active_record/relation/query_methods.rb

```diff
@@ -54,16 +54,17 @@ def not(opts, *rest)
       end
     end
 
+    FROZEN_EMPTY_ARRAY = [].freeze
```

Comment:
Sure, these methods are tested as public; I've added a simple assertion that the return value is frozen.